src/db.rs

```diff
@@ -674,6 +675,32 @@ impl DB {
                      -> Result<(), Error> {
         self.delete_cf_opt(cf, key, &WriteOptions::default())
     }
+
+    pub fn compact_range(&self,
+                         start: Option<&[u8]>,
+                         end: Option<&[u8]>) {
+        unsafe {
+            ffi::rocksdb_compact_range(self.inner,
+                                       opt_bytes_to_ptr(start),
+                                       start.map_or(0, |s| s.len()) as size_t,
+                                       opt_bytes_to_ptr(end),
+                                       end.map_or(0, |e| e.len()) as size_t);
+       }
+    }
+
+    pub fn compact_range_cf(&self,
+                            cf: *mut ffi::rocksdb_column_family_handle_t,
+                            start: Option<&[u8]>,
+                            end: Option<&[u8]>) {
+        unsafe {
+            ffi::rocksdb_compact_range_cf(self.inner,
+                                          cf,
+                                          opt_bytes_to_ptr(start),
+                                          start.map_or(0, |s| s.len()) as size_t,
+                                          opt_bytes_to_ptr(end),
```

Comment:
I feel a little weird about not taking advantage of the [nullable pointer optimization](https://doc.rust-lang.org/book/ffi.html#the-nullable-pointer-optimization) here, but we're not using it elsewhere right now, and I think it would be sort of a departure from how pointers have been handled so far. Maybe in the future it would be better to try to use it, but for now this is fine IMO.